yaksh/views.py

```diff
@@ -869,22 +889,24 @@ def show_all_questions(request):
             form = UploadFileForm(request.POST, request.FILES)
             if form.is_valid():
                 questions_file = request.FILES['file']
-                if questions_file.name.split('.')[-1] == "json":
-                    questions_list = questions_file.read()
-                    question = Question()
-                    question.load_from_json(questions_list, user)
+                file_name = questions_file.name.split('.')
+                if file_name[-1] == "zip":
+                    extract_files(questions_file)
+                    read_json("questions_dump.json", user)
```

Comment:
Is the file in current directory? can we create the files in the /tmp/ to avoid permission issue?